libs/perun/services/src/lib/init-auth.service.ts

```diff
@@ -106,38 +130,108 @@ export class InitAuthService {
 
   /**
    * Handles the auth start. If the configuration property `auto_auth_redirect`
-   * is set to true, a rediret to the oidc server will be made.
+   * is set to true, a redirect to the oidc server will be made.
    * If the property is set to false, a redirect to local page /login will be
    * made.
    */
-  handleAuthStart(): Promise<void> {
+  handleAuthStart(): void {
     if (location.pathname === '/service-access' || sessionStorage.getItem('baPrincipal')) {
       this.serviceAccess = true;
       this.serviceAccessLoginScreen = true;
-      return new Promise<void>((resolve) => {
-        resolve();
-      });
     } else if (this.storeService.get('auto_auth_redirect')) {
       localStorage.setItem('routeAuthGuard', window.location.pathname);
-      return (
-        this.startAuth()
-          // start a promise that will never resolve, so the app loading won't finish in case
-          // of the auth redirect
-          .then(() => new Promise<void>(() => {})) // eslint-disable-line
-      );
+      this.authService.startAuthentication();
     } else {
       this.setLoginScreen(true);
       localStorage.setItem('routeAuthGuard', window.location.pathname);
-      const query = location.search.substr(1).split('&');
+      const query = location.search.substring(1).split('&');
       const queryParams = {};
       for (const param of query) {
         const p = param.split('=');
         queryParams[p[0]] = p[1];
       }
-      return void this.router.navigate(['login'], {
+      void this.router.navigate(['login'], {
         queryParams: queryParams,
         queryParamsHandling: 'merge',
       });
     }
   }
+
+  /**
+   *  Set up check whether access token expires, when it does, it gets automatically refreshed (if possible)
+   */
+  startRefreshToken(isLoggedIn: boolean): void {
```

Comment:
I dunno but I my opinion this method doesn't need isLoggedIn input - you can check right in the method if user is logged in